src/libopenrave/kinbodystatesaver.cpp

```diff
@@ -18,6 +18,121 @@
 
 namespace OpenRAVE {
 
+/// \brief push link to _listNonCollidingLinksWhenGrabbed of grabbed.
+static void _PushLinkToListNonCollidingLinksWhenGrabbed(Grabbed& grabbed,
+                                                        const int linkindex,
+                                                        const KinBody::LinkConstPtr& pLinkRef,
+                                                        const std::vector<KinBody::LinkPtr>& vLinks,
+                                                        const EnvironmentBasePtr& pEnv)
+{
+    if( linkindex < 0 || linkindex >= (int)vLinks.size() ) {
+        RAVELOG_WARN_FORMAT("env=%s, could not restore link '%s' since its index %d is out of range (body num links is %d)", pEnv->GetNameId()%pLinkRef->GetName()%pLinkRef->GetIndex()%vLinks.size());
+    }
+    else {
+        grabbed._listNonCollidingLinksWhenGrabbed.push_back(vLinks.at(linkindex));
```

Comment:
this access is already ranged checked, may as well use `operator[]` instead of `at` here so we don't waste time checking again

Reply:
right!

fixed it!